lib/uklock/Config.uk

```diff
@@ -38,4 +46,18 @@ if LIBUKLOCK
 			Metrics related to mutex objects: current amount of (un)locked
 			objects, as well as number of successful/failed locking attempts
 			since startup.
+
+	config LIBUKLOCK_MUTEX_RECURSE
+		bool "Recursive mutex lock"
+		default n
+		depends on LIBUKLOCK_MUTEX
+		help
+			Enable recursive locking for mutex
+
+	config LIBUKLOCK_RWLOCK_WRITE_RECURSE
+		bool "Recursive writer lock"
+		default n
+		depends on LIBUKLOCK_RWLOCK
+		help
+			Enable recursive writer lock
```

Comment:
I am wondering if it makes sense to globally enable/disable this. Because if you want to use it in a single place this will affect all locks in the system, which might also hide bugs or trigger unexpected or unintended behavior.

Would it be possible to decide on a per-lock basis, when initializing the lock?

Reply:
Yes, we can keep a write recurse flag along with rwlock. This flag can be configured at the time of initialization.